### DISCUSSION TITLE
低代码的实现示例

### DISCUSSION THREAD

**Discussion by HycJack** [CrystalWindSnake/ex4nicegui#198] [Ideas]
请问有低代码的实现示例吗？初步想用nicegui构建一个低代码平台

**Comment by CrystalWindSnake** [comment]
这个话题范围有些大，我觉得至少需要一个界面设计器，可以拖拽生成组件。我认为目前 nicegui 确实可以做到，但还需要周边一些生态支持才行。

**Reply by HycJack** [reply]
我看到作者你封装了draggable组件，就想到可以继续实现，做这个界面设计器，然后导出为模板代码。同时也可以形成组件库或者应用库，现在nicegui的组件库还是太少了。有兴趣，可以一起搞搞

**Reply by CrystalWindSnake** [reply]
draggable组件还是聚焦在数据响应式。我有尝试做界面设计器，感觉难点在于嵌套容器。我有另一个项目 nicegui-toolkit，算是设计器的一个局部功能，是 code first 模式(用户先写代码，通过设计器可视化修改组件各种样式，然后自动作用于代码文件上)。不过目前没有太多时间放到那个项目上
